### PR TITLE
updates micronaut version to 3.5.2

### PR DESCRIPTION
**Pull request by grimmjo**

### CODE REVIEW

**Comment by grimmjo** [comment]
Hey @melix,

after upgrading to 3.5.0 there seems to be a problem with build cache. Do you have any advise?

**Comment by melix** [comment]
This is a nasty bug in Gradle. Let's disable binary compatibility checks for this one.

**Comment by grimmjo** [comment]
Can we make a clean build, e.g. without the cache in github actions? I can reproduce the problem locally.

**Comment by grimmjo** [comment]
gradle/gradle#14220 might be the related issue

**Comment by melix** [comment]
Why do you think the problem is the cache? It's a dependency resolution bug so cleaning the cache won't help.

**Comment by grimmjo** [comment]
My mistake! I just saw the path in the error message last night and thought it would be a caching problem. Next time I should first read the stacktrace 😄

**Review by melix — APPROVED** [review]

**Comment by grimmjo** [comment]
@melix: Can I disable the binary compatibility checks in the build.gradle for example? 🤔 

@graemerocher: Can you tell me who is responsible for merging the PRs in this repo? It looks like you are pretty busy 😃

**Comment by grimmjo** [comment]
@melix: thank you so much for your help =) upgrading to gradle 7.5-rc2 solved the problem. Can you merge this PR?

File: config/accepted-api-changes.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+[
+  {
+    "type": "io.micronaut.nats.reactive.PublishState",
+    "member": "Class io.micronaut.nats.reactive.PublishState",
+    "reason": "PublishState has been removed"
```
Comment (melix) [inline config/accepted-api-changes.json:5]: Small question: if `PublishState` has been removed then shouldn't the version be bumped to 4.0.0?

**Review by melix — APPROVED** [review]

File: config/accepted-api-changes.json (hunk 1 above)
Comment (grimmjo) [reply config/accepted-api-changes.json:5]: PublishState was only used internally. It was holding the informations for publishing a new message and I replaced it with the nats message itself, holding all the information.
As I started this project, I've missed adding `@Internal` to the class.

Comment (melix) [reply config/accepted-api-changes.json:5]: Right, thanks!

**Pull request merged**